src/componets/Clock.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+type Props = {
+  name: string
+};
+
+type State = {
+  time: string;
+};
+
+function getTime() {
+  return (new Date()).toLocaleTimeString();
+}
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    time: getTime(),
+  };
+
+  clockTimer = 0;
+
+  componentDidMount() {
+    this.clockTimer = window.setInterval(() => {
```

Comment:
you don't need a `window` call here
```suggestion
    this.clockTimer = setInterval(() => {
```

Reply:
`Type 'Timer' is not assignable to type 'number'` and `clearInterval(id?: number | undefined)`
so I just left it